osh/hub/osh_xmlrpc/scan.py

```diff
@@ -265,6 +271,10 @@ def find_tasks(request, query):
     if tasks is not None:
         result = list(tasks.order_by("-dt_finished").values_list(
             "id", flat=True))
+        # truncate the result in case `latest` is specified, this reduces the amount of
+        # data transferred over the network
+        if latest:
+            result = [result[0]]
```

Comment:
There are at least two problems with this piece of code:

* While your changes make sure that only a single element is transferred over network, you are still needlessly evaluating the whole `QuerySet` due to the `list()` conversion above.
* The code will crash if the `result` list is empty.

Something like this should fix both issues: 
```suggestion
        result = tasks.order_by("-dt_finished").values_list("id", flat=True))
        if latest:
            first_elem = result.first()
            result = [first_elem] if first_elem is not None else [] 
        else:
            result = list(result)
```

Reply:
I agree that there could be unnecessary evaluation. Amended the patch with the suggested change, thanks for the proposed changes!